components/mpas-framework/src/tools/registry/gen_inc.c

```diff
@@ -1750,20 +1764,21 @@ int parse_var(FILE *fd, ezxml_t registry, ezxml_t superStruct, ezxml_t currentVa
 
 	// Determine name of pointer for this field.
 	set_pointer_name(type, ndims, pointer_name, time_levs);
-	if (time_levs > 1) {
-		fortprintf(fd, "      allocate(%s(%d))\n", pointer_name, time_levs);
-	} else {
-		fortprintf(fd, "      allocate(%s)\n", pointer_name);
-	}
 
-	for(time_lev = 1; time_lev <= time_levs; time_lev++){
-		if (time_levs > 1) {
-			snprintf(pointer_name_arr, 1024, "%s(%d)", pointer_name, time_lev);
+	if (time_levs > 1 || time_levs_from_config) {
```

Comment:
```suggestion
	if (time_levs_from_config || time_levs > 1) {
```
This way, it doesn't matter what `time_levs` has in it when `time_levs_from_config == true`.